muspinsim/input/gipaw.py

```diff
@@ -39,27 +39,31 @@ def __init__(self, file_io: Union[str, IO]):
             file_io = open(file_io, encoding="utf_8")
 
         # Attempt to skip to relevant part of the file
-        while not file_io.readline().strip().startswith("----- total EFG -----"):
-            continue
+        file_valid = False
 
-        loop = True
-        while loop:
+        while line := file_io.readline():
+            if line.strip().startswith("----- total EFG -----"):
+                file_valid = True
+                break
+
+        if not file_valid:
+            raise ValueError(f"Failed to load file GIPAW output file {file_io}")
+
+        while line := file_io.readline():
             # Should be 3 lines at a time per element e.g.
             #      V    1       -0.008879        0.000000       -0.011803
             #      V    1        0.000000       -0.030597        0.000000
             #      V    1       -0.011803        0.000000        0.039476
 
             lines = [
-                file_io.readline().strip(),
+                line.strip(),
                 file_io.readline().strip(),
                 file_io.readline().strip(),
             ]
 
             atom = self._parse_efg(lines)
             if atom is not None:
                 self._atoms.append(atom)
-            else:
-                loop = False
```

Comment:
I haven't tried running the code yet, but should we still have a break condition here? I.E. `else: break`? Looking at the example file you gave me after the `----- total EFG -----` section we have `----- total EFG (symmetrized) -----`, so we would keep reading lines but just fail to assign them to atoms since the header puts us out of sync by one line.

Reply:
Just tested this and it continued to read until the bottom of the file, so while it doesn't break for that file in particular but in general I'd be worried that if we had an extra header somewhere it might line up and give us more atoms than we were expecting, to say nothing of the fact we're spending time reading lines that aren't relevant.